zend/callable.cpp

```diff
@@ -147,9 +152,15 @@ void Callable::initialize(zend_internal_function_info *info, const char *classna
 #else
     // the properties that are available on php 7.2 and higher
     info->required_num_args = _required;
+#if PHP_VERSION_ID < 80000
     info->return_reference = false;
     info->_is_variadic = false;
     info->type = ZEND_TYPE_ENCODE((int)_return, true);
+#else
+    if ((int)_return) {
+        info->type = (zend_type)ZEND_TYPE_INIT_CODE((int)_return, true, 0);
+    }
```

Comment:
type needs to be initialized for `void` functions as well. Otherwise the uninitialized value is used and can lead to early segfaults.

I was able to solve this with
```
else {
        info->type = (zend_type)ZEND_TYPE_INIT_NONE(0);
}
``